src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -18,183 +29,856 @@ object HullmodExoticHandler {
+                }
+            }
+
+            @JvmStatic
+            fun CheckAndRemoveFromAllChildModulesVariants(
+                    fleetMember: FleetMemberAPI,
+                    hullmodExotic: HullmodExotic,
+                    onShouldCallback: OnShouldCallback,
+                    onRemoveFromChildModuleCallback: OnRemoveFromChildModuleCallback
+            ) {
+                // First things first, figure out whether we're running from Refit or Exotica screen
+                val isFromRefitScreen = runningFromRefitScreen()
+                val workMode = if (isFromRefitScreen) {
+                    HullmodExoticHandlerWorkMode.LENIENT
+                } else {
+                    HullmodExoticHandlerWorkMode.STRICT
+                }
+
+                //Lets go through all 'installed modules' and uninstall from them, instead of these module slots
+                // that were the root cause of the whole problem... damn you 3-5am copypasted code ...
+                val hullmodExoticKey = HullmodExoticKey(
+                        hullmodExotic = hullmodExotic,
+                        parentFleetMemberId = fleetMember.id
+                )
+                val currentInstallData = synchronized(lookupMap) { lookupMap[hullmodExoticKey] }
+                val installedOnVariantsList = currentInstallData?.listOfVariantsWeInstalledOn
+                        ?: throw IllegalStateException("No InstallData found for key ${hullmodExoticKey}")
+
+                // Carry on now that we have install data
+
+                // If we have slots, start doing what needs to be done
+                for (installedOnVariant in installedOnVariantsList) {
+                    val mods = ShipModLoader.get(fleetMember, installedOnVariant)
+                    mods?.let { nonNullMods ->
+                        // Since this can be *any* HullmodExotic referencing their own ExoticHullmods, we should first
+                        // check the ExoticHullmodLookup map for any instances of the exotic hullmod.
+                        // And if we find one, we'll just pass it over to the HullmodExoticHandler to remove it from this fleetmember
+                        val shouldRemoveFromModuleVariant = HullmodExoticHandler.shouldRemoveHullmodExoticFromVariant(
+                                hullmodExotic = hullmodExotic,
+                                parentFleetMember = fleetMember,
+                                variant = installedOnVariant,
+                                workMode = workMode
+                        )
+                        onShouldCallback.execute(
+                                onShouldResult = shouldRemoveFromModuleVariant,
+                                moduleVariant = installedOnVariant
+                        )
+                        logger.info("onDestroy()\tshouldRemoveFromModuleVariant: ${shouldRemoveFromModuleVariant}, variant: ${installedOnVariant}")
```

Comment:
1) `log(...)`
2) info is fine, but alot of these logs contents should also be consolidated, since they were literally copypasted from `HullmodExotic`

Reply:
probably done, since there are no explicit calls on `logger` anywhere anymore. resolving.